samza-api/src/main/java/org/apache/samza/operators/MessageStream.java

```diff
@@ -276,13 +282,19 @@
 
   /**
    * Sends messages in this {@link MessageStream} to a {@link Table}. The type of input message is expected
-   * to be {@link KV}, otherwise a {@link ClassCastException} will be thrown.
-   *
+   * to be {@link KV}, otherwise a {@link ClassCastException} will be thrown. After doing a put to table this operator
+   * propagates the message in this {@link MessageStream} to the next chained operator.
+   * <p>
+   *  Note: It is a guarantee that message is first written to table and then only it is propagated to next operator
```

Comment:
What do you mean by "It is a guarantee that message is first written to table". Are you saying that we flush the table before calling next?

"Note: The message will be written but may not be flushed to the underlying table before its propagated to the chained operators. Whether the message can be read back from the Table in the chained operator depends on whether it was flushed and whether the Table offers read after write consistency."

Reply:
What do you mean by "It is a guarantee that message is first written to table". Are you saying that we flush the table before calling next? 
No, just put()

Will do the necessary updates